gui/src/2D/controls/selector.ts

```diff
@@ -0,0 +1,383 @@
+import {Control} from "./control";
+import {TextBlock} from "./textBlock";
+import {Checkbox} from "./checkbox";
+import {RadioButton} from "./radioButton";
+import {Slider} from "./slider";
+
+/**
+ * Class used to store Selector properties
+ */
+export class Selector {
+    /** Text  used for label */
+    private _text: string;
+    /** Function called when selected */ 
+	private _func: () => any; 
+    /** whether checked or not */
+    private _checked: boolean;
+    /** position in array */
+    private _nb: number;
+    
+     /**
+     * Creates a new Selector
+     * @param text is the label for the selector
+     * @param func is the function called when the Selector is checked
+     * @param checked is true when Selector is checked
+     * @param nb is the position of the Selector in the SelectionGroup it is added to
+     */
+    constructor(text:string, func: ()=> any, checked:boolean, nb: number) {
+        this._text = text;
+        this._func = func;
+        this._checked = checked;
+        this._nb = nb;
+    }
+
+    /** Gets the text */
+    public get text(): string {
+        return this._text;
+    }
+    
+    /** Gets the function that is called when checked */
+    public get func(): ()=>any {
+        return this._func;
+    }
+
+    /** Gets the checked value */
+    public get checked(): boolean {
+        return this._checked;
+    }
+    
+    /** Gets the position number */
+    public get nb(): number {
+        return this._nb;
+    }
+}
+
+/**
+ * Class used to store SliderBar properties
+ */
+export class SliderBar{
+    /** Text  used for label */
+    private _text: string;
+    /** Function called when selected */ 
+	private _func: () => any; 
+    /** unit name, eg degrees, metres, etc */
+    private _unit: string;
+    /** Function to format value */ 
+    private _onVal: (v:number)=> any;
+    /** Minimum of value range */
+    private _min: number;
+    /** Maximum of value range */
+    private _max: number;
+    /** starting value */
+	private _value: number;
+    
+     /**
+     * Creates a new SliderBar
+     * @param text is the label for the SliderBar
+     * @param func is the function called when the Slider moves
+     * @param unit is a string describing the units used, eg degrees or metres
+     * @param onVal is the function used to format the value displayed, eg radians to degrees
+     * @param min is the minimum value for the Slider
+     * @param max is the maximum value for the Slider
+     * @param value is the start value for the Slider between min and max
+     */
+    constructor(text:string, func: ()=> any, unit: string, onVal: (v:number)=> any, min: number, max: number, value: number) {
+        this._text = text;
+        this._func = func;
+        this._unit = unit;
+        this._onVal = onVal;
+        this._min = min;
+        this._max = max;
+        this._value = value;
+    }
+
+    /** Gets the text */
+    public get text(): string {

```

Comment:
We should make these properties read/write

Reply:
This also goes to the question of the degree of customisation to allow a user I address in my comments on headers and controls. Under the current structure setters are not needed. What I think you are saying is that after the construction of a selection panel a user should be able to change the the labels and functions that apply to a given control and so the properties for a selector need to be writable as well. Personally I think it would be easier for the user to dispose of the current panel and create a new one.  For added customisation you could also think about allowing the user to delete and add controls to a group after construction. What do you think about this for a compromise I add in the setters for later development and for the moment add in only customisation via a theme as suggested in my comments below.

Reply:
The thing is I can foresee an usage where I have dozens of controls and recreating everything to change a label seems overkill. I think we could easily have two arrays: one for the headers and one from the controls and we can just expose this two arrays. Then users can do panel.headers[10].text = ...

Reply:
OK will go with that and work on it.

Reply:
You rock